sigar_linux.go

```diff
@@ -155,6 +155,36 @@ func (self *FileSystemList) Get() error {
 	return err
 }
 
+func (self *DiskList) Get() error {
+	diskList := make(map[string]DiskIo)
+
+	err := readFile(Procd+"/diskstats", func(line string) bool {
+		fields := strings.Fields(line)
+		if len(fields) < 13 {
+			return true
+		}
+		majorDevId, _ := strtoull(fields[0])
+		if !isBlockDevice(majorDevId) {
+			return true
+		}
+		io := DiskIo {}
+		deviceName := fields[2]
+		io.ReadOps, _ = strtoull(fields[3])
+		readBytes, _ := strtoull(fields[5])
+		io.ReadBytes = readBytes * 512
+		io.ReadTimeMs, _ = strtoull(fields[6])
+		io.WriteOps, _ = strtoull(fields[7])
+		writeBytes, _ := strtoull(fields[9])
+		io.WriteBytes = writeBytes * 512
+		io.WriteTimeMs, _ = strtoull(fields[10])
+		io.IoTimeMs, _ = strtoull(fields[12])
```

Comment:
The convention in gosigar is to return 0s for values we can't parse. We could opt to not return anything for invalid lines instead.